File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-6.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int l=0;
        int r=0;
        int total=0;
        int minWindow=Integer.MAX_VALUE;
        while (r<nums.length){
            total+=nums[r];
            while (total>=target){
                if (total>=target){
                    int window=r-l+1;
                    minWindow=Math.min(minWindow,window);
                }
                total-=nums[l];
                l++;
                
            }
            r++;
        }
        if (minWindow<Integer.MAX_VALUE){
            return minWindow;        
        }
        return 0;
    }
}

File: Data Structures & Algorithms/target-sum/submission-6.py
class Solution:
    def findTargetSumWays(self, nums: List[int], target: int) -> int:
        total={}
        if nums[0]==0:
            total[0]=2
        else:
            total[nums[0]]=1
            total[-nums[0]]=1
        cnt=0
        for i in range(1,len(nums)):
            num=nums[i]
            temp={}
            for key, val in total.items():
                k1=key+num
                k2=key-num
                temp[k1]=temp.get(k1,0)+val
                temp[k2]=temp.get(k2,0)+val
            total=temp
        print(total)
        if target not in total:
            return 0
        return total[target]
                